qiskit/providers/models/pulsedefaults.py

```diff
@@ -94,44 +101,325 @@ class Command(BaseModel):
-        cmd_def (list[Command]): Backend command definition.
+    """Description of default settings for Pulse systems. These are operations or settings that
+    may be good starting points for the Pulse user. The user may modify these defaults through
+    the provided methods to build a reference to custom operations, which may in turn be used
+    for building Schedules or converting circuits to Schedules.
     """
 
-    def __init__(self, qubit_freq_est, meas_freq_est, buffer,
-                 pulse_library, cmd_def, **kwargs):
-        self.qubit_freq_est = qubit_freq_est
-        self.meas_freq_est = meas_freq_est
+    def __init__(self,
+                 qubit_freq_est: List[float],
+                 meas_freq_est: List[float],
+                 buffer: int,
+                 pulse_library: List[PulseLibraryItem],
+                 cmd_def: List[Command],
+                 **kwargs: Dict[str, Any]):
+        """
+        Validate and reformat transport layer inputs to initialize this.
+
+        Args:
+            qubit_freq_est: Estimated qubit frequencies in GHz.
+            meas_freq_est: Estimated measurement cavity frequencies in GHz.
+            buffer: Default buffer time (in units of dt) between pulses.
+            pulse_library: Pulse name and sample definitions.
+            cmd_def: Operation name and definition in terms of Commands.
+            **kwargs: Other attributes for the super class.
+        """
+        super().__init__(**kwargs)
+
         self.buffer = buffer
+        self._qubit_freq_est_ghz = qubit_freq_est
+        self._meas_freq_est_ghz = meas_freq_est
+        self._qubit_freq_est_hz = [freq * 1e9 for freq in qubit_freq_est]
+        self._meas_freq_est_hz = [freq * 1e9 for freq in meas_freq_est]
+        # TODO: These should be massaged for the user
         self.pulse_library = pulse_library
         self.cmd_def = cmd_def
 
-        super().__init__(**kwargs)
+        # The processed and reformatted circuit operation definitions
+        self._ops_def = defaultdict(dict)
+        # A backwards mapping from qubit to supported operation
+        self._qubit_ops = defaultdict(list)
+
+        # Build the above dictionaries from pulse_library and cmd_def
+        self.converter = QobjToInstructionConverter(pulse_library, buffer)
+        for op in cmd_def:
+            qubits = _to_tuple(op.qubits)
+            self._qubit_ops[qubits].append(op.name)
+            pulse_insts = [self.converter(inst) for inst in op.sequence]
+            self._ops_def[op.name][qubits] = ParameterizedSchedule(*pulse_insts, name=op.name)
+
+    @property
+    def qubit_freq_est(self) -> List[float]:
+        """
+        Return the estimated resonant frequency for the given qubit in Hz.
+
+        Returns:
+            The frequency of the qubit resonance in Hz.
+        """
+        warnings.warn("The qubit frequency estimation was previously in GHz, and now is in Hz.")
+        return self._qubit_freq_est_hz
```

Comment:
Up to 2 months (maybe less if we release more frequently). Open to suggestions

Reply:
One alternative idea would be to create a `qubit_freq_est_hz` and then raise a deprecation warning for the GHz `qubit_freq_est` (but would want `qubit_freq_est` to eventually be Hz). Since Pulse is in beta, I thought it would be appropriate to use a quicker deprecation strategy

Reply:
@1ucian0